libmscore/album.cpp

```diff
@@ -0,0 +1,1489 @@
+    }
+    m_combinedScore->setTextMovement(true);
+
+    // add the album's scores as movements and layout the combined score
+    for (auto& item : m_albumItems) {
+        m_combinedScore->addMovement(item->score());
+    }
+    if (m_drawFrontCover) {
+        updateFrontCover();
+    }
+    if (m_generateContents) {
+        m_combinedScore->setfirstRealMovement(2);
+        updateContents();
+    } else {
+        m_combinedScore->setfirstRealMovement(1);
+    }
+    m_combinedScore->setLayoutAll();
+    m_combinedScore->update();
+
+    //
+    // parts - excerpts
+    //
+    if (m_combinedScore->excerpts().isEmpty()) {
+        for (auto& e : m_albumExcerpts) {
+            //
+            // prepare Excerpts
+            //
+            Excerpt* ne = new Excerpt(m_combinedScore.get());
+            ne->setTitle(e->title);
+            for (auto& partIndex : e->partIndices) {
+                ne->parts().append(m_combinedScore->parts().at(partIndex));
+            }
+            ne->setTracks(e->tracks);
+            //
+            // create Excerpts
+            //
+            MasterScore* nscore = new MasterScore(ne->oscore());
+            ne->setPartScore(nscore);
+            nscore->setName(ne->oscore()->title() + "_part_" + ne->oscore()->excerpts().size());
+            m_combinedScore->addExcerpt(ne);
+            Excerpt::createExcerpt(ne);
+
+            // a new excerpt is created in AddExcerpt, make sure the parts are filed
+            for (auto& ee : ne->oscore()->excerpts()) {
+                if (ee->partScore() == nscore && ee != ne) {
+                    ee->parts().clear();
+                    ee->parts().append(ne->parts());
+                }
+            }
+
+            for (auto& m : *m_combinedScore->movements()) {
+                if (m == m_combinedScore.get()) {
+                    continue;
+                }
+                Excerpt* ee = createMovementExcerpt(prepareMovementExcerpt(ne, m));
+                nscore->addMovement(static_cast<MasterScore*>(ee->partScore()));
+            }
+
+            nscore->setLayoutAll();
+            nscore->undoChangeStyleVal(MSQE_Sid::Sid::spatium, 25.016); // hack: normally it's 25 but it draws crazy stuff with that
+                                                                        // if you disable this the ScoreView becomes Picasso
+            nscore->update();
+        }
+    }
+
+    return m_combinedScore.get();
+}
+
+//---------------------------------------------------------
+//   getCombinedScore
+//---------------------------------------------------------
+
+MasterScore* Album::getCombinedScore() const
+{
+    return m_combinedScore.get();
+}
+
+//---------------------------------------------------------
+//   loadFromFile
+///     Load a .msca or .album file.
+//---------------------------------------------------------
+
+bool Album::loadFromFile(const QString& path, bool legacy)
+{
+    std::cout << "Loading album from file..." << std::endl;
```

Comment:
make this a `qDebug`?

Reply:
or a `std::cerr` at least. But indeed better a `qDebug()`

Reply:
I'll delete it, it's useless.